src/utils/node_byte_utils.ts

```diff
@@ -124,8 +124,8 @@ export const nodeJsByteUtils = {
     return Buffer.from(hex, 'hex');
   },
 
-  toHex(buffer: Uint8Array): string {
-    return nodeJsByteUtils.toLocalBufferType(buffer).toString('hex');
+  toHex(buffer: NodeJsBuffer, start?: number, end?: number): string {
```

Comment:
```suggestion
  toHex(buffer: Uint8Array, start?: number, end?: number): string {
```

The idea is that we do not know we have a NodeBuffer or a Uint8Array upon entering this function, so we call toLocalBufferType which is no-op-ish if we already have one, and converts (without copying) it to a NodeBuffer if it isn't one.

Reply:
Ah I think I switched it to play around with something and forgot to change it back Thanks for catching that.